packages/core-utils/src/query.js

```diff
@@ -163,13 +163,32 @@ function isParamApplicable(paramInfo, query, config) {
   return true;
 }
 
+/**
+ * Helper method which replaces OTP flex modes with single FLEX mode that's
+ * more useful and easier to work with.
+ */
+export function reduceOtpFlexModes(modes) {
+  return modes.reduce((prev, cur) => {
+    const newModes = prev;
+    // Add the current mode if it is not a flex mode
+    if (!cur.includes("FLEX")) {
+      newModes.push(cur);
+      // If it is a flex mode, do not add it but rather add the custom flex mode
+      // if not already present
+    } else if (!newModes.includes("FLEX")) {
+      newModes.push("FLEX");
```

Comment:
Sorry, these were the lines that made me think of `Set`, but that's just a code refinement.